### PR TITLE
monitoring: alert owners

### PR DESCRIPTION
**Pull request by bobheadxi**
Adds a new required `Observable.Owner` field, with predefined owners based on [2021 engineering org](https://about.sourcegraph.com/handbook/engineering/2021_org), as proposed in https://github.com/sourcegraph/sourcegraph/issues/12010 for RFC-189

Many of these teams don't exist yet, but it seems the ones that don't are clearly covered by an existing team in the team page (for example, [backend infrastructure](https://about.sourcegraph.com/handbook/engineering/backend-infrastructure) => current Cloud team)

Owners were assigned based on guesses, please let me know if any should change!

### CODE REVIEW

File: monitoring/generator.go [hunk 1]
```diff
@@ -104,6 +104,25 @@ func (r Row) validate() error {
 	return nil
 }
 
+// ObservableOwner denotes a team that owns an Observable. The current teams are described in
+// the handbook: https://about.sourcegraph.com/handbook/engineering/2021_org
+type ObservableOwner string
+
+const (
+	// Core Products teams
```
Comment (unknwon) [inline monitoring/generator.go:112]:
```suggestion
	// Core products teams
```

Unless you copied from handhook?

File: monitoring/generator.go [hunk 2]
```diff
@@ -104,6 +104,25 @@ func (r Row) validate() error {
...
+	ObservableOwnerSearch               ObservableOwner = "search"
+	ObservableOwnerCampaigns            ObservableOwner = "campaigns"
+	ObservableOwnerCodeIntel            ObservableOwner = "code-intel"
+	ObservableOwnerExtensibility        ObservableOwner = "extensibility"
+	ObservableOwnerCodeHostIntegrations ObservableOwner = "code-host-integrations"
+
+	// Core Services teams
```
Comment (unknwon) [inline monitoring/generator.go:119]:
```suggestion
	// Core services teams
```

File: monitoring/generator.go (hunk 1 above)
Comment (bobheadxi) [reply monitoring/generator.go:112]: it's from [this page](https://about.sourcegraph.com/handbook/engineering/2021_org)

image

Comment (bobheadxi) [reply monitoring/generator.go:112]: I like the lower case though

**Comment by bobheadxi** [comment]
Is this a flakey test?

image

**Pull request merged**